src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/AmendVideoBookingService.kt

```diff
@@ -0,0 +1,96 @@
+package uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service
+
+import jakarta.persistence.EntityNotFoundException
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.client.locationsinsideprison.LocationValidator
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.client.prisonersearch.PrisonerValidator
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.entity.VideoBooking
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.model.Prisoner
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.model.request.AmendVideoBookingRequest
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.model.request.BookingType
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.repository.CourtRepository
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.repository.PrisonAppointmentRepository
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.repository.PrisonRepository
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.repository.ProbationTeamRepository
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.repository.VideoBookingRepository
+import java.time.LocalDateTime
+
+@Service
+class AmendVideoBookingService(
+  private val courtRepository: CourtRepository,
+  private val probationTeamRepository: ProbationTeamRepository,
+  private val videoBookingRepository: VideoBookingRepository,
+  prisonAppointmentRepository: PrisonAppointmentRepository,
+  prisonRepository: PrisonRepository,
+  locationValidator: LocationValidator,
+  prisonerValidator: PrisonerValidator,
+) : CreateVideoAppointmentService(prisonRepository, prisonAppointmentRepository, locationValidator, prisonerValidator) {
+  companion object {
+    private val log = LoggerFactory.getLogger(this::class.java)
+  }
+
+  @Transactional
+  fun amend(videoBookingId: Long, request: AmendVideoBookingRequest, amendedBy: String): Pair<VideoBooking, Prisoner> {
+    val booking = videoBookingRepository.findById(videoBookingId)
```

Comment:
Was wondering here if we could just get away with add a repo method and do away with the require.  The UI won't allow a different type though really.

`fun findByVideoBookingIdAndBookingType(id: Long, bookingType: String)`

It is either found or not found. Just a thought.